src/database/sql_database.cc

```diff
@@ -607,35 +607,33 @@ std::unique_ptr<std::vector<int>> SQLDatabase::getServiceObjectIDs(char serviceP
 template <class En>
 std::string SQLDatabase::parseSortStatement(const std::string& sortCrit, const std::vector<std::pair<std::string, En>>& keyMap, const std::map<En, std::pair<std::string, std::string>>& colMap)
 {
-    std::ostringstream sortSql;
-    if (!sortCrit.empty()) {
-        bool firstEntry = true;
-        for (auto&& seg : splitString(sortCrit, ',')) {
-            if (firstEntry) {
-                sortSql << ", ";
-                firstEntry = false;
-            }
-            seg = trimString(seg);
-            bool desc = (seg[0] == '-') ? true : false;
-            if (seg[0] == '-' || seg[0] == '+') {
-                seg = seg.substr(1);
-            } else {
-                log_warning("Unknown sort direction {} in {}", seg, sortCrit);
-            }
-            bool defined = false;
-            for (auto&& [tag, key] : keyMap) {
-                if (tag == seg) {
-                    sortSql << TQD(colMap.at(key).first, colMap.at(key).second);
-                    sortSql << seg << (desc ? " DESC" : "");
-                    defined = true;
-                }
-            }
-            if (!defined) {
-                log_warning("Unknown sort key {} in {}", seg, sortCrit);
+    if (sortCrit.empty()) {
+        return "";
+    }
+    std::vector<std::string> sort;
+    for (auto&& seg : splitString(sortCrit, ',')) {
+        seg = trimString(seg);
+        bool desc = (seg[0] == '-') ? true : false;
+        if (seg[0] == '-' || seg[0] == '+') {
+            seg = seg.substr(1);
+        } else {
+            log_warning("Unknown sort direction '{}' in '{}'", seg, sortCrit);
+        }
+        bool defined = false;
```

Comment:
loop seems inefficient. that if statement can be taken out I think.

Reply:
The sortCriteria looks like "+dc::title,-upnp::track_number". the first if checks for the signs (+/-) the second for the key (title etc.). Which could be moved outside without losing the context.

Reply:
Something like:

```
auto it = std::find_if(xxx)
if (it != x.end())
```

Reply:
Valid point. Sometimes I'm going too straight. Thanks.